From aec2efd9c8484ab0d87deeecee1cedf510ea98ba Mon Sep 17 00:00:00 2001
From: Olivier Moysan <olivier.moysan@foss.st.com>
Date: Wed, 20 Mar 2024 17:42:38 +0100
Subject: [PATCH 1011/1141] clk: stm32mp25: allow set_parent rate on mdf clock

Allow the MDF to change its parent clock rate, to manage rate potential
conflicts with SAI.

Signed-off-by: Olivier Moysan <olivier.moysan@foss.st.com>
Change-Id: Ib41e0b54d8719295726362fd4b8dbd7b1e69e393
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/368446
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Gabriel FERNANDEZ <gabriel.fernandez@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 drivers/clk/stm32/clk-stm32mp25.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/clk/stm32/clk-stm32mp25.c b/drivers/clk/stm32/clk-stm32mp25.c
index 1277e91f9e51..72d55012a3cf 100644
--- a/drivers/clk/stm32/clk-stm32mp25.c
+++ b/drivers/clk/stm32/clk-stm32mp25.c
@@ -782,7 +782,7 @@ static CLK_STM32_COMPOSITE(ck_mco2, mco2_src, 0, GATE_MCO2, MUX_MCO2, NO_STM32_D
 
 /* MDF */
 static CLK_STM32_GATE(ck_icn_p_mdf1, "ck_icn_ls_mcu", 0, GATE_MDF1);
-static CLK_STM32_GATE(ck_ker_mdf1, "ck_flexgen_23", 0, GATE_MDF1);
+static CLK_STM32_GATE(ck_ker_mdf1, "ck_flexgen_23", CLK_SET_RATE_PARENT, GATE_MDF1);
 
 /* OSPI */
 static CLK_STM32_GATE(ck_icn_p_ospiiom, "ck_icn_ls_mcu", 0, GATE_OSPIIOM);
-- 
2.39.2

